### PR TITLE
RDMA_CM support in signature and pipelining examples

### PR DESCRIPTION
**Pull request by ghost**
Some users asked about how to use RDMA_CM with the DV signature APIs because it is not clear. So, let's update our examples to use RDMA_CM.

### CODE REVIEW

File: dv_signature_offload_api_example/sig_example.c [hunk 1]
```diff
@@ -136,10 +131,9 @@ struct msg_t {
 
 /* structure of system resources */
 struct resources {
-	struct ibv_device_attr device_attr;
-	/* Device attributes */
-	struct ibv_port_attr port_attr;
-	struct cm_con_data_t remote_props; /* values to connect to remote side */
+	struct rdma_cm_id *cm_id;	/* connection on client side,*/
+					/* listener on service side. */
```
Comment (EugeneKochetov) [inline dv_signature_offload_api_example/sig_example.c]: service -> server

File: dv_signature_offload_api_example/sig_example.c [hunk 2]
```diff
@@ -512,91 +381,104 @@ static int post_send(struct resources *res, int opcode, const struct msg_t *req)
...
+static void set_sig_domain_crc32(struct mlx5dv_sig_block_domain *domain, void *sig)
 {
 	struct mlx5dv_sig_crc *crc = sig;
 
 	memset(domain, 0, sizeof(*domain));
 	memset(crc, 0, sizeof(*crc));
 
 	domain->sig_type = MLX5DV_SIG_TYPE_CRC;
-	domain->block_size = (config.block_size == 512) ? MLX5DV_BLOCK_SIZE_512 : MLX5DV_BLOCK_SIZE_4096;
+        domain->block_size = (config.block_size == 512)
```
Comment (EugeneKochetov) [inline dv_signature_offload_api_example/sig_example.c]: indentation mismatch

File: dv_signature_offload_api_example/sig_example.c [hunk 3]
```diff
@@ -1054,427 +746,233 @@ static int resources_create(struct resources *res)
...
-			res->ib_ctx = NULL;
-		}
-
-		if (dev_list) {
-			ibv_free_device_list(dev_list);
-			dev_list = NULL;
-		}
+	/* signature specific attributes */
+	mlx5_qp_attr.comp_mask = MLX5DV_QP_INIT_ATTR_MASK_SEND_OPS_FLAGS;
+	mlx5_qp_attr.send_ops_flags = MLX5DV_QP_EX_WITH_MKEY_CONFIGURE;
```
Comment (EugeneKochetov) [inline dv_signature_offload_api_example/sig_example.c]: Duplicated code

File: dv_signature_offload_api_example/sig_example.c [hunk 4]
```diff
@@ -1054,427 +746,233 @@ static int resources_create(struct resources *res)
...
+	if (!res->recv_mr)
+		goto err_exit;
+
+	res->data_mr = alloc_mr(res->pd, res->data_buf_size);
+	if (!res->data_mr)
+		goto err_exit;
+
+	res->pi_mr = alloc_mr(res->pd, res->pi_buf_size);
+	if (!res->pi_mr)
+		goto err_exit;
```
Comment (EugeneKochetov) [inline dv_signature_offload_api_example/sig_example.c]: pi_buf_size is zero in interleaved mode. Won't it fail here?

File: dv_signature_offload_api_example/sig_example.c [hunk 5]
```diff
@@ -1993,35 +1640,43 @@ int main(int argc, char *argv[])
...
+	rc = cm_bind(&res);
+	if (rc)
+		goto free_res_and_exit;
+
 	rc = resources_create(&res);
 	if (rc) {
-		fprintf(stderr, "failed to create resources: %s\n", strerror(rc));
-		if (rc == EOPNOTSUPP || rc == ENOTSUP)
+		if (skip) {
+			err("Signature pipelining feature is not supported by the specified RDMA device\n");
```
Comment (EugeneKochetov) [inline dv_signature_offload_api_example/sig_example.c]: This is not a pipelining example

File: dv_signature_offload_api_example/sig_example.c (hunk 1 above)
Comment (ghost) [reply dv_signature_offload_api_example/sig_example.c]: done

File: dv_signature_offload_api_example/sig_example.c (hunk 2 above)
Comment (ghost) [reply dv_signature_offload_api_example/sig_example.c]: Fixed

File: dv_signature_offload_api_example/sig_example.c (hunk 3 above)
Comment (ghost) [reply dv_signature_offload_api_example/sig_example.c]: Removed

File: dv_signature_offload_api_example/sig_example.c (hunk 4 above)
Comment (ghost) [reply dv_signature_offload_api_example/sig_example.c]: Added "if (res->pi_buf_size)" here.

File: dv_signature_offload_api_example/sig_example.c (hunk 5 above)
Comment (ghost) [reply dv_signature_offload_api_example/sig_example.c]: Rephrased

**Review by EugeneKochetov — APPROVED** [review]

**Pull request merged**